src/lib/structures/ty.mli

```diff
@@ -48,20 +48,19 @@ type t =
   (** Abstract types applied to arguments. [Text (args, s)] is
       the application of the abstract type constructor [s] to
       arguments [args]. *)
+
   | Tfarray of t * t
   (** Functional arrays. [TFarray (src,dst)] maps values of type [src]
       to values of type [dst]. *)
-  | Tsum of Uid.t * Uid.t list
-  (** Enumeration, with its name, and the list of its constructors. *)
 
-  | Tadt of Uid.t * t list
-  (** Application of algebraic data types. [Tadt (a, params)] denotes
+  | Tadt of Uid.t * t list * bool
```

Comment:
Unnamed booleans should generally be avoided; if we keep this flag, we should either `adt_kind = Adt | Enum` and use that instead of `bool`, use `[` Adt | `Enum ]`, or define `Tadt` with an inline record with an `is_enum : bool` field.

Reply:
Sure. I didn't make much effort here as we plan to get rid of the legacy types soon. I will create an extra ADT.